### PR TITLE
plasmahud: fix runtime errors and other cleanup

### LINKED ISSUE #209750: PlasmaHUD not working (missing Gtk namespace)
### Describe the bug
Plasma HUD is not working due to missing Gtk namespace. I'm using nixos-unstable channel.

### Steps To Reproduce
1.  Install `pkgs.plasma-hud` by modifying `/etc/nixos/configurations.nix`.

### Expected behavior
PlasmaHUD should trigger `rofi` to filter and select an option, then execute it.

### Additional context
If I run `plasma-hud` from CLI, it shows
```console
[user@system:~]$ plasma-hud
Traceback (most recent call last):
  File "/nix/store/5x7ydziw4b3m4fxs6hnqkic3la5jr989-plasma-hud-19.10.1/bin/.plasma-hud-wrapped", line 5, in <module>
    gi.require_version("Gtk", "3.0")
  File "/nix/store/1psjx5nf2gglgd59cjrnqkqpxyy5qy9p-python3.10-pygobject-3.42.2/lib/python3.10/site-packages/gi/__init__.py", line 126, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Gtk not available
```

### Notify maintainers
@pasqui23 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.3, NixOS, 23.05 (Stoat), 23.05pre440068.a518c771485`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

- fix runtime errors (fixes #209750)
- clean up `installPhase`
- fix scoping for `propagatedBuildInputs`
- formatting
- clean up `src`
- remove `with lib;`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eclairevoyant** [comment]
@asd-noor Can you check if this works now? I no longer see the error, but not sure how to test this if it's actually working either.

**Review by nicoonoclaste — CHANGES_REQUESTED** [review]

File: pkgs/misc/plasma-hud/default.nix [hunk 1]
```diff
@@ -1,42 +1,54 @@
...
+  propagatedBuildInputs = (with python3.pkgs; [
     dbus-python
+    pygobject3
     setproctitle
     xlib
-    pygobject3
-    gobject-introspection
-  ];
-  format = "other";
+  ]) ++ [ rofi ];
```
[the remark this answers is not in the document]
Comment (eclairevoyant) [reply pkgs/misc/plasma-hud/default.nix:31]: I'm aware, but I would rather be explicit.
Frankly I'd just remove the `with` entirely, if this was my own code.

File: pkgs/misc/plasma-hud/default.nix [hunk 2]
```diff
@@ -1,42 +1,54 @@
...
   version = "19.10.1";
+  format = "other";
 
   src = fetchFromGitHub {
     owner = "Zren";
-    repo = pname;
+    repo = "plasma-hud";
     rev = version;
-    sha256 = "19vlc156jfdamw7q1pc78fmlf0h3sff5ar3di9j316vbb60js16l";
+    hash = "sha256-1AQtgVlrmzBkim1kVZzTAwJHq0OH3YAPr6o5aUpgdKc=";
```
[the remark this answers is not in the document]
Comment (eclairevoyant) [reply pkgs/misc/plasma-hud/default.nix:18]: That's why there's multiple commits, to separate the fix vs other refactoring.

**Review by nicoonoclaste — APPROVED** [review]
LGTM, sorry for missing the multiple commits the first time around  :sweat:

**Review by superherointj — APPROVED** [review]

**Pull request merged**